server/src/main/java/org/cloudfoundry/identity/uaa/oauth/UaaAuthorizationRequestManager.java

```diff
@@ -387,8 +385,10 @@ public TokenRequest createTokenRequest(Map<String, String> requestParameters, Cl
     }
 
     protected Set<String> extractScopes(Map<String, String> requestParameters, ClientDetails clientDetails) {
-        boolean clientCredentials = "client_credentials".equals(requestParameters.get(OAuth2Utils.GRANT_TYPE)) || 
-                JWT_BEARER_GRANT.equals(requestParameters.get(OAuth2Utils.GRANT_TYPE));
+        boolean clientCredentials = 
+                OauthGrant.CLIENT_CREDENTIALS.equalsIgnoreCase(requestParameters.get(OAuth2Utils.GRANT_TYPE)) ||
```

Comment:
Change all comparisons to equals instead of equalsIgnoreCase to be consistent with RFC 6749.